src/Dialect/ONNX/ONNXOps/Additional/LayoutTransform.cpp

```diff
@@ -43,10 +43,18 @@ LogicalResult ONNXLayoutTransformOp::inferShapes(
 //===----------------------------------------------------------------------===//
 LogicalResult ONNXLayoutTransformOp::verify() {
   if (hasShapeAndRank(getData()) && hasShapeAndRank(getOutput())) {
-    if (getShape(getData().getType()) != getShape(getOutput().getType()))
-      return emitOpError("Input and output tensors must have the same shape");
+    // Get the unknown dimension from data.
+    auto dataType = getData().getType().dyn_cast<RankedTensorType>();
+    auto outputType = getOutput().getType().dyn_cast<RankedTensorType>();
+    // Check if input has a static dimension and output has dynamic dimensions
+    if (!dataType && outputType) {
```

Comment:
This check means `dataType` is unranked (e.g. `tensor<2xf32>`) and `outputType` is ranked (e.g. `tensor<?x3x?xf32>`). I guess what you want to check is `dataType` or `outputType` is unranked, that is `if (!dataType || !outputType)` or `!(dataType && outputType)`. How about this check:

```C
if (auto dataType = getData().getType().dyn_cast<RankedTensorType>())
  if (auto outputType = getOutput().getType().dyn_cast<RankedTensorType>())
     return (getShape(dataType) == getShape(outputType);

return success();
```

Reply:
Thanks for your feedback...let me try it out and see my results.